kb/0002-scripting-overview.md

```diff
@@ -40,6 +40,461 @@ The following list is sorted alphabetically and is based on the latest build of
 * `program\weather\` - weather system definitions and usage.
 * `program\worldMap\` - everything world map related.
 
+## Typing
+
+In order to be compiled, the scripting files should contain good C-like code. This also means that the variables should be strongly typed. Commonly used are base C types: `int`, `float`, but also `bool` and a simple to use `string`. Any custom structs are supported in the form of `object`s. Last, arrays are supported of the above.
+
+### Base Types
+
+There're three base types in scripting language: 
+
+* `int`: a signed 32-bit integer
+
+``` C++
+syntax: 
+	int var_name;
+examples:
+	int GameTime = -2053; // OK
+	int FreeSlots = 75;   // OK
+```
+
+* `float`: a single-precision floating-point number. Because there're no `double`s, specifying `f` at the end is not required but allowed. 
+
+``` C++
+syntax: 
+	float var_name;
+examples:
+	float WindSpeed = 2.0;         // OK
+	float WindDirection = -0.0057; // OK
+	float ShipDirection = 3.14f;   // OK
+```
+
+* `bool`: a logical `true`/`false` operator where `true` is any non-zero value and `false` is zero.
+
+``` C++
+syntax: 
+	bool var_name;
+examples:
+	bool IsSpeaking = true;                  // OK
+    bool IsDrunk = (AlcoholInBlood > 10.0f); // OK
```

Comment:
Indentation in not consistent here and in the `strings` section

Reply:
I didn't realize that in the original documentation tabs were used instead of spaces. Fixed!